### ISSUE TITLE
FXAA breaks with Resolution Scale

### ISSUE THREAD

**Issue by mrjustaguy** [godotengine/godot#68096]
### Godot version

4.0 Beta 3

### System information

Windows 11, Vulkan, GTX 1050 Ti 516.94

### Issue description

FXAA seems to calculate Pixel size by Resolution, without taking into account Resolution Scale, which creates artifacts especially clear at very low resolution scales.

### Steps to reproduce

Turn on FXAA
1) Set Resolution Scale to 0.5 - Observe result
2) Set Resolution Scale to 1.0 and set Viewport to Half Res - Observe result
Observe how cases 1&2 clearly don't have the same result with FXAA

**Comment by mrjustaguy** [comment]
It seems pixel_size is giving the Viewport Resolution, which is not modified by Render Scale, which is the case for other effects using it too, such as DOF

**Comment by Calinou** [comment]
> which is the case for other effects using it too, such as DOF

See https://github.com/godotengine/godot/issues/56911.

**Comment by Calinou** [comment]
See https://github.com/godotengine/godot/issues/108722#issuecomment-3085412744 for a comparison.

I've attempted to do this, but it makes no visual difference whatsoever in both Forward+ and Mobile, both on FXAA and SMAA: https://github.com/Calinou/godot/tree/fxaa-smaa-perform-at-internal-resolution

Any ideas?

**Comment by mrjustaguy** [comment]
Not really, I don't understand Godot's internals well enough, though https://github.com/RGDTAB/Godot-SMAA may provide some insights into why it worked when this doesn't, though I doubt it given that it's using a different path with the Compositor
